### PR TITLE
[ Master - Bug 12681 ] - In agent ticket search, profile field is not modernize.

### PR DESCRIPTION
**Pull request by s7design**
https://bugs.otrs.org/show_bug.cgi?id=12681

Hello @dvuckovic ,

Hereby is proposal for fixing reporters issue. Implemented 'Modernize' class for SearchProfile field. This was causing issues in past since Core.UI.Dialog.js is calling FocusFirstElement() function which is finding first field and placing focus on it. When that field is modernize, it auto-expand showing available selectable options. Included additional check to not place focus if found field is part of modernize field 'input'.

Additionally added 'id' and 'for' as parts of label for 'Search template'. This way focus is on field when label is clicked.

Please let me know if there are any issues or questions regarding this PR.

Regards,
Sanjin

### CODE REVIEW

**Comment by s7design** [comment]
Hello @dvuckovic ,

PR is updated with proposal for fixing navigation and focus on hidden field. In case when modernize field is found in dialog as first element, new field is created with same parent before modernize field and focus is set on it. This field have position: absolute and z-index: -1 which is allowing it to remain behind modernize selection field and serve as 'dummy' focus field.

Please let me know if there are any issues or questions regarding this commit. 

Regards,
Sanjin

File: var/httpd/htdocs/skins/Agent/default/css/Core.Dialog.css [hunk 1]
```diff
@@ -157,6 +157,11 @@
     width: 98%;
 }
 
+.Dialog .Field .FocusField {
+    position: absolute;
+    z-index: -1;
+}
+
```
Comment (mrcbnsls) [inline var/httpd/htdocs/skins/Agent/default/css/Core.Dialog.css]: I think there might be cases where just having a z-index of -1 is not enough, e.g. when the element which is on top of the "hidden" element has no background color. In that case the "hidden" element would still be visible in a weird way. We should better add positioning, too (similar to ``.InvisibleText``, ``left: -9999px`` and ``top: -9999px``).

File: var/httpd/htdocs/js/Core.UI.Dialog.js [hunk 2]
```diff
@@ -88,10 +88,17 @@ Core.UI.Dialog = (function (TargetNS) {
...
-        $('div.Dialog:visible .Content')
-            .find('a:visible, input:visible, textarea:visible, select:visible, button:visible')
-            .filter(':first')
-            .focus();
+        var FirstElement = $('div.Dialog:visible .Content').find('a:visible, input:visible, textarea:visible, select:visible, button:visible').filter(':first'),
+            FocusField;
+
+        // Check if found first element is modernize input field, prepend input field which is behind modernize field and set focus on it allowing navigation.
+        if (FirstElement.closest('.Field').find('.Modernize').length) {
+            FocusField = FirstElement.parent();
```
Comment (mrcbnsls) [inline var/httpd/htdocs/js/Core.UI.Dialog.js]: I didn't try the code out, but it seems wrong or at least unnecessarily complex to me. If a ``.Field`` container contains more than one field of which one is of type ``.Modernize``, the hidden field would be added without any need. Why not just check if ``FirstElement.hasClass('InputField_Search')``?

File: Kernel/Output/HTML/Templates/Standard/AgentTicketSearch.tt [hunk 3]
```diff
@@ -15,7 +15,7 @@
         <input type="hidden" name="ShownAttributes" value="" id="ShownAttributes">
         <fieldset class="TableLike">
             <legend><span>[% Translate("Templates") | html %]</span></legend>
-            <label>[% Translate("Search template") | html %]:</label>
+            <label for="SearchProfile" id="LabelSearchProfile">[% Translate("Search template") | html %]:</label>
```
Comment (mrcbnsls) [inline Kernel/Output/HTML/Templates/Standard/AgentTicketSearch.tt]: What's the use of the ``id`` here? I think ``for``should be enough, no?

File: Kernel/Modules/AgentTicketSearch.pm [hunk 4]
```diff
@@ -1815,21 +1815,13 @@ sub Run {
...
             UserLogin => $Self->{UserLogin},
         );
-        delete $Profiles{''};
-        delete $Profiles{'last-search'};
-        if ($EmptySearch) {
-            $Profiles{''} = '-';
-        }
-        else {
-            $Profiles{'last-search'} = '-';
-        }
```
Comment (mrcbnsls) [inline Kernel/Modules/AgentTicketSearch.pm]: Is this part no longer needed?

**Comment by dvuckovic** [comment]
Merged in d8fe430 and 48f0009. Thanks to @s7design for pull request and @mrcbnsls for insights.

**Pull request closed without merging**